src/workspaceState.ts

```diff
@@ -0,0 +1,73 @@
+import * as vscode from 'vscode'
+
+type WorkspaceStateKey = 'LastActiveManifestUri' | 'Initialized' | 'DependenciesUpdated' | 'DependenciesBuilt' | 'ApplicationBuilt'
+
+export class WorkspaceState {
+    private readonly extCtx: vscode.ExtensionContext
+
+    constructor(extCtx: vscode.ExtensionContext) {
+        this.extCtx = extCtx
+    }
+
+    async resetPipelineState(): Promise<void> {
+        await this.setInitialized(false)
+        await this.setDependenciesUpdated(false)
+        await this.setDependenciesBuilt(false)
+        await this.setApplicationBuilt(false)
+    }
+
+    async setContext(contextName: string, value: boolean): Promise<void> {
```

Comment:
Should this be private?

Reply:
this is also used by manifestManager when a manifest is found

Reply:
ahh, we can just update the context here and make it private